src/Type.hs

```diff
@@ -204,7 +207,7 @@ noStorageRead store expr = for_ (keys store) $ \name ->
 -- ensures that key types match value types in an U.Assign
 checkAssign :: Env -> U.Assign -> Err [StorageUpdate]
 checkAssign env@Env{contract,store} (U.AssignVal (U.StorageVar _ (StorageValue (FromAbi typ)) name) expr)
-  = sequenceA [_Update (Item typ contract name []) <$> inferExpr env expr]
+  = sequenceA [_Update (_Item contract name []) <$> checkExpr env typ expr]
```

Comment:
I'm probably missing something obvious, but what is an `_Item` and where is it defined?

Reply:
These underscore definitions are a shorthand for some language constructors (`Item`, `Loc`, `TExp`, and others) that automatically retrieve the `SType` (here `typ`) when there is a `SingI` instance, and the user does not have to provide it. 

(These already existed before the PR, I just changed this line to use `_Item`, as in other places.)

https://github.com/ethereum/act/blob/227f74a73c0897a2a84600bebc2a81127ae764c1/src/Syntax/TimeAgnostic.hs#L163-L164


Here, because a is an `AType`, it has a corresponding singleton, which is retrieved by the typeclass method `sing`:

https://github.com/ethereum/act/blob/227f74a73c0897a2a84600bebc2a81127ae764c1/src/Syntax/Types.hs#L64-L66